### PR TITLE
V4 - added new parent BaseAnatomy class

### PR DESCRIPTION
**Pull request by kalisp**
## Brief description
Reason is to get info directly from project_doc, not from settings.

## Testing notes:
1. start with this step
2. follow this step

### CODE REVIEW

File: openpype/pipeline/anatomy.py [hunk 1]
```diff
@@ -339,6 +335,59 @@ def replace_root_with_env_key(self, filepath, template=None):
...
+        self._site_name = site_name
+        project_info = self.get_project_data_and_cache(project_name, site_name)
+
+        super(Anatomy, self).__init__(
+            project_info["project_doc"],
+            project_info["local_settings"]
+        )
+
+    @classmethod
+    def get_project_data_and_cache(cls, project_name, site_name):
```
Comment (iLLiCiTiT) [inline openpype/pipeline/anatomy.py:360]: I would not cache `local_settings`, they're already cached in settings logic. This should just return project document.

File: openpype/pipeline/anatomy.py [hunk 2]
```diff
@@ -103,20 +103,16 @@ def _prepare_anatomy_data(anatomy_data):
...
-                        value_queue.append(value)
+        apply_local_settings_on_anatomy_settings(anatomy_data,
+                                                 local_settings, project_name)
 
-                    elif isinstance(value, six.string_types):
-                        item[key] = value.replace("{task}", "{task[name]}")
-        return anatomy_data
+        self._data = anatomy_data
 
     def reset(self):
```
Comment (iLLiCiTiT) [inline openpype/pipeline/anatomy.py:111]: Reset is not used anywhere I would remove it.

File: openpype/pipeline/anatomy.py [hunk 3]
```diff
@@ -39,34 +49,21 @@ def __init__(self, roots):
...
-        if not project_name:
-            project_name = os.environ.get("AVALON_PROJECT")
-
-        if not project_name:
-            raise ProjectNotSet((
-                "Implementation bug: Project name is not set. Anatomy requires"
-                " to load data for specific project."
-            ))
-
+    def __init__(self, project_doc, local_settings):
```
Comment (iLLiCiTiT) [inline openpype/pipeline/anatomy.py:60]: I think `site_name` is required to handle local settings content.

File: openpype/pipeline/anatomy.py [hunk 4]
```diff
@@ -338,6 +327,199 @@ def replace_root_with_env_key(self, filepath, template=None):
...
+        if not project_name:
+            project_name = os.environ.get("AVALON_PROJECT")
+
+        if not project_name:
+            raise ProjectNotSet((
+                "Implementation bug: Project name is not set. Anatomy requires"
+                " to load data for specific project."
+            ))
+
+        self._site_name = site_name
```
Comment (iLLiCiTiT) [inline openpype/pipeline/anatomy.py:453]: I would move this to `BaseAnatomy`.

File: openpype/pipeline/anatomy.py [hunk 5]
```diff
@@ -338,6 +327,199 @@ def replace_root_with_env_key(self, filepath, template=None):
...
 
+    def _project_doc_to_anatomy_data(self, project_doc):
+        """Convert project document to anatomy data.
+
+        Probably should fill missing keys and values.
+        """
+        if not project_doc:
+            return {}
+
+        project_settings_root = ProjectSettings(
```
Comment (iLLiCiTiT) [inline openpype/pipeline/anatomy.py:338]: Why it's using `ProjectSettings`?

**Comment by kalisp** [comment]
Fixed according to comments.

File: openpype/pipeline/anatomy.py [hunk 6]
```diff
@@ -338,6 +328,163 @@ def replace_root_with_env_key(self, filepath, template=None):
         data = self.root_environmets_fill_data(template)
         return rootless_path.format(**data)
 
+    def _project_doc_to_anatomy_data(self, project_doc):
+        """Convert project document to anatomy data.
+
+        Probably should fill missing keys and values.
+        """
+        if not project_doc:
+            return {}
```
Comment (iLLiCiTiT) [inline openpype/pipeline/anatomy.py:337]: It would crash much earlier if it would be true.
```suggestion
```

**Review by iLLiCiTiT — APPROVED** [review]
Looks good and works. One last suggestion otherwise approved.

**Pull request merged**